splice_machine/src/test/java/com/splicemachine/db/impl/sql/compile/PredicateTupleIT.java

```diff
@@ -0,0 +1,170 @@
+import com.splicemachine.homeless.TestUtils;
+import org.junit.*;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.spark_project.guava.collect.Lists;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLSyntaxErrorException;
+import java.sql.Types;
+import java.util.Collection;
+import java.util.List;
+
+import static org.junit.Assert.*;
+import static org.hamcrest.Matchers.containsString;
+
+/**
+ * Test predicate with tuples
+ */
+@RunWith(Parameterized.class)
+public class PredicateTupleIT extends SpliceUnitTest {
+
+    private Boolean useSpark;
+
+    @Parameterized.Parameters
+    public static Collection<Object[]> data() {
+        Collection<Object[]> params = Lists.newArrayListWithCapacity(2);
+        params.add(new Object[]{true});
+        params.add(new Object[]{false});
+        return params;
+    }
+    private static final String SCHEMA = PredicateTupleIT.class.getSimpleName();
+
+    @ClassRule
+    public static SpliceSchemaWatcher schemaWatcher = new SpliceSchemaWatcher(SCHEMA);
+
+    @ClassRule
+    public static SpliceWatcher classWatcher = new SpliceWatcher(SCHEMA);
+
+    @Rule
+    public SpliceWatcher methodWatcher = new SpliceWatcher(SCHEMA);
+
+    @BeforeClass
+    public static void createSharedTables() throws Exception {
+        TestUtils.executeSqlFile(classWatcher.getOrCreateConnection(), "subquery/PredSimplTestTables.sql", "");
+    }
+
+    @AfterClass
+    public static void dropUDF() throws Exception {
+        try {
+            TestUtils.executeSqlFile(classWatcher.getOrCreateConnection(), "subquery/PredSimplTestCleanup.sql", "");
+        }
+        catch (Exception e) {
+            // Don't error out if the UDF we want to drop does not exist.
+        }
+    }
+
+    public PredicateTupleIT(Boolean useSpark) {
+        this.useSpark = useSpark;
+    }
+
+
+    private void testQuery(String sqlText, String expected) throws Exception {
+        ResultSet rs = null;
+        try {
+            rs = methodWatcher.executeQuery(sqlText);
+            assertEquals("\n" + sqlText + "\n", expected, TestUtils.FormattedResult.ResultFactory.toStringUnsorted(rs));
+        }
+        finally {
+            if (rs != null)
+                rs.close();
+        }
+    }
+
+    @Test
+    public void testSimpleTuple() throws Exception {
+        String query = format("select * from A --SPLICE-PROPERTIES useSpark=%s\n" +
+                "where (a1, a2) = (1,10)", useSpark);
+        String expected = "A1 |A2 |A3  |\n" +
+                "-------------\n" +
+                " 1 |10 |100 |";
+
+        testQuery(query, expected);
+    }
+
+    @Ignore("Find a way to pass nulls inside a tuple")
+    @Test
+    public void testTupleWithNulls() throws Exception {
+        String query = format("select count(*) from D --splice-properties useSpark=%s\n" +
+                "where (d1,d2,d3) = (null,null,null)", useSpark);
```

Comment:
You may use the expression `case when 1=0 then 1 end` to generate a null value.

Reply:
Removing this test as it does not make much sense to compare to null